bob-apps/bin/bobd.rs

```diff
@@ -120,6 +148,115 @@ async fn main() {
     }
 }
 
+fn configure_testmode(sub_matches: &ArgMatches) -> Result<(ClusterConfig, NodeConfig), String> {
+    let mut addresses = Vec::with_capacity(1);
+    let port = match sub_matches.value_of("grpc-port") {
+        Some(v) => v.parse().unwrap(),
+        None => 20000
+    };
+    let mut this_node = None;
+    if let Some(node_list) = sub_matches.value_of("nodes") {
+        let available_ips: Vec<String> = NetworkInterface::show().unwrap().into_iter().filter_map(|itf|
+            match itf.addr.unwrap() {
+                Addr::V4(addr) => {
+                    Some(addr.ip.to_string())
+                },
+                _ => None
+        }).collect();
+
+        for (addr, index) in node_list.split(",").zip(0..usize::MAX) {
+            let split = &addr.split_once(":").unwrap();
+            if this_node.is_none() {
+                for ip in available_ips.iter() {
+                    if ip == split.0 && port == split.1.parse::<u16>().unwrap() {
+                        this_node = Some(index);
+                        break;
+                    }
+                }
+            }
+            addresses.push(String::from(addr));
+        }
+        if this_node.is_none() {
+            return Err(String::from("No available IPs found"));
+        }
+    } else {
+        this_node = Some(0);
+        addresses.push(format!("127.0.0.1:{}", port))
+    }
+    let cluster = ClusterConfig::get_testmode(
+        sub_matches.value_of("data").unwrap_or("data").to_string(),
+        addresses).unwrap();
+    let http_api_port = sub_matches.value_of("restapi-port").and_then(|v|  Some(v.parse().unwrap()));
+    let node = cluster.get_testmode_node(this_node.unwrap(), http_api_port).unwrap();
+
+    init_testmode_logger(log::LevelFilter::Info);
+
+    check_folders(&node, true);
+
+    println!("Bob has started");
```

Comment:
Bob actually has not started at this point. I suggest to write `Bob starting`

Reply:
Fix: [f0b0215](https://github.com/qoollo/bob/pull/588/commits/f0b02157ff45022f0859f4807d820eeae9b256c0)